src/features/session_replay/aggregate/index.js

```diff
@@ -152,6 +162,11 @@ export class Aggregate extends AggregateBase {
       else return
     }
 
+    // If an error was noticed before the mode could be set (like in the early lifecycle of the page), immediately set to FULL mode
+    if (this.mode === MODE.ERROR && this.errorNoticed) {
+      this.mode = MODE.FULL
+    }
+
```

Comment:
This already happens in L107-L111

Reply:
This extra `errorNoticed` check is needed for the case where:

An error was thrown and noticed before `mode` could be set to "ERROR" (ie before SR loads, before the sampling)